src/printer/smt2/smt2_printer.cpp

```diff
@@ -472,13 +518,9 @@ void Smt2Printer::toStream(std::ostream& out,
 
   bool stillNeedToPrintParams = true;
   bool forceBinary = false; // force N-ary to binary when outputing children
-  bool parametricTypeChildren = false;   // parametric operators that are (op t1 ... tn) where t1...tn must have same type
-  bool typeChildren = false;  // operators (op t1...tn) where at least one of t1...tn may require a type cast e.g. Int -> Real
   // operator
-  Kind k = n.getKind();
   if (n.getNumChildren() != 0 && k != kind::INST_PATTERN_LIST
-      && k != kind::APPLY_TYPE_ASCRIPTION && k != kind::CONSTRUCTOR_TYPE
-      && k != kind::CAST_TO_REAL)
```

Comment:
I remember I added `CAST_TO_REAL` here to avoid extra parentheses for `CAST_TO_REAL` which is hidden from the user. I guess this is no longer needed since the printer is faithful to the AST.

Reply:
Yes, that's right.  I simplified this, which notice also applies to APPLY_TYPE_ASCRIPTION (which is a cast for datatypes).